### PR TITLE
Add the Pimoroni Servo 2040 board

### PR DESCRIPTION
**Pull request by Nashenas88**
This PR adds the board support package and two simple examples.
* Animates a rainbow wheel across the RGB LEDs on the board.
* Moves a servo that's connected to GPIO0 back and forth.

### CODE REVIEW

File: boards/pimoroni-servo2040/src/lib.rs
Comment (Nashenas88) [inline boards/pimoroni-servo2040/src/lib.rs:52]: All of the pins named servo* are connected to standard servo pinouts (signal, V+, Gnd) and are wired to 180Ohm resistor (see https://cdn.shopify.com/s/files/1/0174/1800/files/servo2040_schematic.pdf?v=1648817752 and https://cdn.shopify.com/s/files/1/0174/1800/files/servo-2040-drawing.png?v=1649088626 for layout along the top). Should I remove the aliases for simplicity since they're unlikely to be used by the other names?

File: boards/pimoroni-servo2040/src/lib.rs
Comment (Nashenas88) [inline boards/pimoroni-servo2040/src/lib.rs:18]: Do these constants belong here? They seem very useful given the setup of the board.

**Comment by Nashenas88** [comment]
Are there any other files that need to be modified when adding a new board?

**Comment by ithinuel** [comment]
`README.md` at the root of the repo also has entries for the BSPs.

**Review by ithinuel — APPROVED** [review]
LGTM. I don't have the board to test it.

No strong opinion about the constants nor aliases.

**Comment by Nashenas88** [comment]
Would it help if I upload a video of both examples?

**Comment by 9names** [comment]
It's not necessary, but if you don't mind i'd love to see a video of the examples in action.
Would it be okay if I added them to the readme after this pr is merged, like I did for
[makerpi_rp2040](https://github.com/9names/makerpi_rp2040)?

**Comment by Nashenas88** [comment]
Sure, I'll record some tomorrow morning, I'm not in any rush to merge it in, so it could be included with this PR or after. I don't have merge approval, so feel free to merge when you feel comfortable. My only open question is whether the current aliases on the servo pins make sense.

File: boards/pimoroni-servo2040/src/lib.rs
Comment (9names) [reply boards/pimoroni-servo2040/src/lib.rs:50]: The aliases are always optional.
That said, I think 3 of them still make sense for a servo depending on how you're controlling it (PWM/PIO0/PIO1)

File: boards/pimoroni-servo2040/src/lib.rs
Comment (9names) [reply boards/pimoroni-servo2040/src/lib.rs:18]: Constants here are fine :D

**Comment by Nashenas88** [comment]
servo-2040-rainbow

Here's the rainbow animation. I'll show the servo in a little bit. Need to set up the power supply. When I'm home I can also show the oscilloscope output showing the 50Hz frequency on the servo waveform.

**Pull request merged**
